### DISCUSSION TITLE
.env file values are undefined

### DISCUSSION THREAD

**Discussion by corrieuys** [vercel/next.js#21967] [Help]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

12.18.1

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 10

**How are you deploying your application?**

Github push + vercel cli deploy

**Describe the Bug**

In development, next doesnt seem to load .env files correctly. I am currently putting all my environment variables in my next config file, but this is not secure, and I want to move to proper env variables.

Here is my workflow:

1. Add environment variables in project settings (dev, preview, prod).
2. "vercel env pull" in the CLI in VS Code. A new .env file is created, with the values.
3. npm run dev. The terminal confirms that the env file has been loaded. (Loaded env from C:\Users\.....\.env)

**Expected Behavior**

I console log one of the variables (Firebase API Key) high up in the app. Here there are two different scenarios:
- A: with the next.config file, the value is displayed correctly in the console, and the terminal window.
- B: with the .env file, the value is 'undefined' in the browser console log, BUT, the value is displayed correctly in the terminal log.

What is going on here? Why would the terminal display correctly, if the value is undefined?

Next says it loads the .env file, but the values are not available in the app. Does the existence of the next.config file influence the .env loading? I currently load PWA settings and Image component settings from next.config, so I do need it. Its just not secure to put environment vars in there.

**To Reproduce**

As above. I added the variables in the project settings on Vercel, and pulled the variables through the CLI. 
Restart the app with npm run dev. 
Next confirms the .env file is loaded. 
Values are undefined.

**Comment by f0wu5u** [comment]
@corrieuys read here
https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser
